kube/src/api/metadata.rs

```diff
@@ -33,16 +31,16 @@ where
         self.metadata()
     }
 
-    fn name(&self) -> String {
-        self.meta().name.clone().expect("kind has metadata.name")
+    fn name(&self) -> &str {
+        self.meta().name.as_ref().expect("kind has metadata.name")
     }
 
-    fn resource_ver(&self) -> Option<String> {
-        self.meta().resource_version.clone()
+    fn resource_ver(&self) -> Option<&str> {
+        self.meta().resource_version.as_ref().map(|rv| rv.as_ref())
     }
 
-    fn namespace(&self) -> Option<String> {
-        self.meta().namespace.clone()
+    fn namespace(&self) -> Option<&str> {
+        self.meta().namespace.as_ref().map(|ns| ns.as_ref())
```

Comment:
`&Option<String>` vs `Option<&str>` awkwardness ^

Reply:
Should be `self.meta().namespace.as_deref()`

Reply:
That is nicer. Thanks!